src/main/java/seedu/address/logic/commands/room/EditRoomCommand.java

```diff
@@ -0,0 +1,220 @@
+            + "Example: " + COMMAND_WORD + "23 "
+            + PREFIX_ROOM_NUMBER + "123 "
+            + PREFIX_PATIENT_NAME + "Mary Doe";
+
+    public static final String MESSAGE_EDIT_ROOM_SUCCESS = "Edited Room: %1$s";
+    public static final String MESSAGE_ROOM_NOT_EDITED = "At least one field to edit must be provided.";
+    public static final String MESSAGE_DUPLICATE_ROOM = "This room already exists in the application.";
+
+    private final Integer roomNumberToEdit;
+    private final EditRoomDescriptor editRoomDescriptor;
+
+    /**
+     * Constructs an EditRoomCommand to edit the room with the room number {@code Integer}.
+     *
+     * @param roomNumberToEdit room number to edit
+     * @param editRoomDescriptor details to edit the room with
+     */
+    public EditRoomCommand(Integer roomNumberToEdit, EditRoomDescriptor editRoomDescriptor) {
+        requireAllNonNull(roomNumberToEdit, editRoomDescriptor);
+        this.roomNumberToEdit = roomNumberToEdit;
+        this.editRoomDescriptor = new EditRoomDescriptor(editRoomDescriptor);
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        List<Room> lastShownList = model.getModifiableRoomList().getRoomObservableList();
+        Index index = model.getModifiableRoomList().checkIfRoomPresent(roomNumberToEdit);
+
+        if (index.getZeroBased() == 0) {
+            throw new CommandException(MESSAGE_INVALID_ROOM_NOT_FOUND);
+        }
+
+        Room roomToEdit = lastShownList.get(index.getZeroBased() - 1);
+        Room editedRoom = createEditedRoom(model, roomToEdit, editRoomDescriptor);
+
+        if (!roomToEdit.isSameRoom(editedRoom) && model.containsRoom(editedRoom)) {
+            throw new CommandException(MESSAGE_DUPLICATE_ROOM);
+        }
+
+        model.setSingleRoom(roomToEdit, editedRoom);
+        return new CommandResult(String.format(MESSAGE_EDIT_ROOM_SUCCESS, editedRoom));
+    }
+
+    /**
+     * Creates and returns a {@code Room} with the details of {@code roomToEdit}
+     * edited with {@code editRoomDescriptor}.
+     *
+     * @param model Current model
+     * @param roomToEdit Room that is to be edited.
+     * @param editRoomDescriptor Details to edit the room with.
+     * @return Room that has been edited.
+     */
+    private static Room createEditedRoom(Model model, Room roomToEdit,
+                                         EditRoomDescriptor editRoomDescriptor) throws CommandException {
+        assert (roomToEdit != null);
+        int updatedRoomNumber = editRoomDescriptor.getRoomNumber().orElse(roomToEdit.getRoomNumber());
+        boolean clearRoom = (editRoomDescriptor.getIsOccupied().orElse(null) != null); //non-null just means clear room
+        boolean hasNewPatient = (editRoomDescriptor.getPatientName().orElse(null) != null);
+        boolean alreadyOccupied = roomToEdit.isOccupied();
+
+        if (clearRoom && !hasNewPatient) {
```

Comment:
Mabybe using a switch could be better for readability?

Reply:
Might be difficult to switch here since there are multiple conditions.

I think the current structure is not too bad. Maybe we can rename `clearRoom` to `isNewEmptyRoom` for clarity? `clearRoom` sounds like the name of a method hahaha.

Also, I think `isCurrentlyOccupied` sounds more like a boolean. How about we use that instead of `alreadyOccupied`?

Reply:
i'm not sure how to use a switch here because the if cases are all different, maybe you want to give me some suggestions?

Reply:
yup sure! i will change it